[IMP] mail_channel_notify_tag: Welcome mail_channel_notify_tag T#54296

rolandojduartem opened the pull request:
This module re-enables channel_ids from the front end when the channels
are mentioned in the messages and then, they are handled to post the
message in the respective channels and an email to the members of the
channels.

The context 'mentioned_channel_ids' is created to handle the channels in
the inheritances.

channel_ids functionality was removed on:
- https://github.com/odoo/odoo/commit/e5338146
- https://github.com/odoo/odoo/commit/59d6bb23

**Record:**

![record](https://github.com/Vauxoo/addons-vauxoo/assets/90422721/c625c5af-6cd6-4605-ae6e-07f567c2fab0)

**Channel:**

![channel](https://github.com/Vauxoo/addons-vauxoo/assets/90422721/4e5c0367-effb-490c-983a-eb30005e70bf)

**Email:**

![email](https://github.com/Vauxoo/addons-vauxoo/assets/90422721/4a6949b2-ee2b-41c0-b35f-48c0f61117ff)

rolandojduartem commented:
Hi, @luisg123v, @oscarolar, @imanie383, @tomeyro and @desdelinux, please, could you review this PR?

For functional review, you can use the runbot of [this MR](https://git.vauxoo.com/vauxoo/instance/-/merge_requests/1623).

oscarolar commented on mail_channel_notify_tag/README.rst:
```diff
@@ -0,0 +1,10 @@
+Mail Channel Notify Tag
+=======================
+
+This module re-enable channel_ids from the front end when the channels
```
```suggestion
This module re-enables channel_ids from the front end when the channels
```

oscarolar commented on mail_channel_notify_tag/README.rst:
```diff
@@ -0,0 +1,10 @@
+
+This module re-enable channel_ids from the front end when the channels
+are mentioned in the messages and then, they are handled to post the
+message in the respective channels and an email to their members.
```
```suggestion
message in the respective channels and an email to its members.
```

oscarolar commented on mail_channel_notify_tag/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "Mail Channel Notify Tag",
+    "summary": "Module to notify channel when they are mentioned in the chatter",
```
```suggestion
    "summary": "Module to notify a channel when they are mentioned in the chatter",
```

oscarolar commented on mail_channel_notify_tag/views/mail_templates.xml:
```diff
@@ -0,0 +1,21 @@
+    <template id="mail_channel_message_repost" name="Mail Channel Message Repost">
+        <div style="width: 100%; margin: 0 auto;">
+            <div style="margin: 5px 0;">
+                New message on:
```
This header is redundant, when we get a message we know we have a "New message on:"

Please remove,

oscarolar replied:
Additionally, I would preserve the original format:

```python
{{SenderName}} - {{Time}} on {{RelatedDocument}}
```

example:

<img width="1006" alt="image" src="https://github.com/Vauxoo/addons-vauxoo/assets/3836433/a2d996d4-15f7-47eb-8537-52549e5d004d">

rolandojduartem replied:
Is this a message in channel?

oscarolar replied:
this is an example in v14

<img width="1616" alt="image" src="https://github.com/Vauxoo/addons-vauxoo/assets/3836433/9468c658-58ad-4342-84cf-01daf75b06d9">

rolandojduartem replied:
The problem is those messages had channel_ids, but we lost that reference, but I will try to replicate that behavior. That is why we sent a message directly to the channel in the message_post.

rolandojduartem replied:
Hi, as we discussed, I think this is not possible because the messages are not sent from the records to the channels anymore, actually the suggested format is still native [here](https://github.com/odoo/odoo/blob/2c39f6cb2e984df459c649569e5ecd3bfeb01169/addons/mail/static/src/components/message/message.xml#L117), but we are sending the message in the channels directly, so originThread is the same channel and it gives no more information, but I made and improvement to the template.

![Screenshot from 2023-09-19 20-11-27](https://github.com/Vauxoo/addons-vauxoo/assets/90422721/a2ce579b-c2ca-43bb-9dec-5f0d4f9caf9c)

oscarolar replied:
Just a small question, is it possible to change the link to be aligned with the text of the subject to maximize space usage, something like:

```
 Asunto Re: INV/2023/00001 en INV/2023/00001
```
![7ca60lik](https://github.com/Vauxoo/addons-vauxoo/assets/3836433/79d2b5cf-02ae-463a-9ef7-3db1edf58386)


If not, just leave it as it is

oscarolar commented on mail_channel_notify_tag/views/mail_templates.xml:
```diff
@@ -0,0 +1,21 @@
+                    t-out="source.display_name"
+                />
+            </div>
+            <div style="background-color: #714B67; margin-bottom: 20px; height: 2px;" />
```
Please remove this line also

rolandojduartem commented:
Hi, @oscarolar, the changes are done.

oscarolar commented:
@luisg123v LGTM

cc @rolandojduartem

luisg123v commented on mail_channel_notify_tag/LICENSE:
```diff
@@ -0,0 +1,27 @@
+Odoo Proprietary License v1.0
```
This is open source, license should be LGPL-1

rolandojduartem replied:
LGPL-3

luisg123v commented on mail_channel_notify_tag/models/mail_followers.py:
```diff
@@ -0,0 +1,13 @@
+    def _get_recipient_data(self, records, message_type, subtype_id, pids=None):
+        """Inheritance to take into account the partners from the channels to notify them by email"""
+        channel_ids = self.env.context.get("mentioned_channel_ids")
+        if channel_ids and pids is not None:
```
Is it possible for `pids` to be `None`? Shouldn't this work also in such case?

luisg123v commented on mail_channel_notify_tag/models/mail_thread.py:
```diff
@@ -0,0 +1,24 @@
+
+
+class MailThread(models.AbstractModel):
+
```
Nitpick:

Remove extra blank line.

Same in the other model

luisg123v commented on mail_channel_notify_tag/models/mail_thread.py:
```diff
@@ -0,0 +1,24 @@
+        if not channel_ids:
+            return
+        kwargs.pop("partner_ids", [])  # Remove target partners to not send duplicate messages.
+        if kwargs.get("subtype_xmlid"):
```
```
subtype_xmlid = kwargs.pop(..., None)
if subtype_xmlid:
    ...
```

luisg123v commented on mail_channel_notify_tag/views/mail_templates.xml:
```diff
@@ -0,0 +1,21 @@
+            </div>
+            <br />
+            <div>
+                <t t-out="message.body" />
```
Combine with outer element

luisg123v commented on mail_channel_notify_tag/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+            "mail_channel_notify_tag/static/src/models/composer_view/composer_view.esm.js",
+        ],
+        "mail.assets_discuss_public": [
+            "mail_channel_notify_tag/static/src/models/composer_view/composer_view.esm.js",
```
Is it expected having this duplicated?

rolandojduartem replied:
Yes, composer view is loaded in the both assets [here](https://github.com/odoo/odoo/blob/9d5b57596836e63eb4d127151bf10dfa79186fa5/addons/mail/__manifest__.py#L83)

luisg123v commented on mail_channel_notify_tag/README.rst:
```diff
@@ -0,0 +1,11 @@
+Mail Channel Notify Tag
+=======================
+
+This module re-enables channel_ids from the front end when the channels
```
`channel_ids` is a technical term, this should be understandable to a non-technical person.

luisg123v commented on mail_channel_notify_tag/README.rst:
```diff
@@ -0,0 +1,11 @@
+
+channel_ids functionality was removed on:
+- https://github.com/odoo/odoo/commit/e5338146
+- https://github.com/odoo/odoo/commit/59d6bb23
```
Add sections:
- Contributors (include myself as planner/reviewer)
- Maintainer

rolandojduartem commented:
Hi, @luisg123v , please, could you review the changes?

rolandojduartem commented:
Hi, @luisg123v, friendly reminder

luisg123v commented on mail_channel_notify_tag/README.rst:
```diff
@@ -0,0 +1,60 @@
+systems and bases its entire operation strategy in the use
+of Open Source Software and its main product is odoo.
+
+To contribute to this module, please visit http://www.vauxoo.com.
```
https

luisg123v replied:
Applied by you, as it's a tiny change and it's the only pending one.

rolandojduartem committed:
[IMP] mail_channel_notify_tag: Welcome mail_channel_notify_tag

This module re-enables channel_ids from the front end when the channels
are mentioned in the messages and then, they are handled to post the
message in the respective channels and an email to the members of the
channels.

The context 'mentioned_channel_ids' is created to handle the channels in
the inheritances.

channel_ids functionality was removed on:
- https://github.com/odoo/odoo/commit/e5338146
- https://github.com/odoo/odoo/commit/59d6bb23

rolandojduartem commented:
Hi, @luisg123v, please, could you merge this PR manually?
